analytics/event.go

```diff
@@ -0,0 +1,31 @@
+package analytics
+
+import "time"
+
+// Holds the name and properties of an analytics event.
+type Event struct {
+	// The name of the event.
+	name string
+
+	// The timestamp of the event. Set to nil to use the current time.
+	timestamp time.Time
+
+	// Custom properties of the event.
+	properties map[string]any
+}
+
+// Returns a new event with the given name, properties, and timestamp. Given no timestamp is provided, the current time is used.
+func NewEvent(name string, properties map[string]any, timestamp ...time.Time) *Event {
```

Comment:
Feels like a bit of an abuse of a variadic function. I'm uncomfortable that if more than one timestamp is given, those extra arguments are silently discarded. Why not just have the caller provide `time.Now()` explicitly? Or, if that's the common case, maybe have a `NewEvent(string, map[string]any)` and a `NewEventWithTime(string, map[string]any, time.Time)`. Or, alternatively, declare a `NewEvent(string, map[string]any)` and a `func (*Event) SetTime(time.Time) *Event` so that we can write `myEvent := NewEvent(...).SetTime(t)`.

Reply:
I've updated this in [8ad54fe](https://github.com/akitasoftware/akita-libs/pull/152/commits/8ad54fe616e1834a3c31cfcd30f0fa36fe03a1e3)